modules/aggregation-layers/src/heatmap-layer/heatmap-layer.ts

```diff
@@ -398,98 +396,101 @@ export default class HeatmapLayer<
     const weightsScale = floatTargetSupport ? 1 : 1 / 255;
-    this.setState({textureSize, format, type, weightsScale});
+    this.setState({textureSize, format, weightsScale});
     if (!floatTargetSupport) {
       log.warn(
-        `HeatmapLayer: ${this.id} rendering to float texture not supported, fallingback to low precession format`
+        `HeatmapLayer: ${this.id} rendering to float texture not supported, falling back to low precision format`
       )();
     }
   }
 
-  getShaders(type) {
-    return super.getShaders(
-      type === 'max-weights-transform'
-        ? {
-            vs: vsMax,
-            _fs: fsMax
-          }
-        : {
-            vs: weightsVs,
-            _fs: weightsFs
-          }
-    );
-  }
-
-  _createWeightsTransform(shaders = {}) {
+  _createWeightsTransform(shaders: {vs: string, fs?: string}) {
     let {weightsTransform} = this.state;
-    weightsTransform?.delete();
+    const {weightsTexture} = this.state;
+    const attributeManager = this.getAttributeManager()!;
 
+    weightsTransform?.destroy();
     weightsTransform = new TextureTransform(this.context.device, {
       id: `${this.id}-weights-transform`,
+      // attributes: this.getAttributes() as any, // TODO(donmccurdy): DO NOT SUBMIT.
+      bufferLayout: attributeManager.getBufferLayouts(),
       vertexCount: 1,
-      targetTexture: this.state.weightsTexture!,
+      targetTexture: weightsTexture!,
       targetTextureVarying: 'weightsTexture',
+      targetTextureChannels: 4,
+      parameters: {
+        depthCompare: 'always',
+        blendColorOperation: 'add',
+        blendColorSrcFactor: 'one',
+        blendColorDstFactor: 'one'
+      },
       ...shaders
     } as TextureTransformProps);
+
     this.setState({weightsTransform});
   }
 
   _setupResources() {
+    // TODO(donmccurdy): DO NOT SUBMIT.
+    console.log('HeatmapLayer#_setupResources()');
+
     this._createTextures();
+    const {device} = this.context;
     const {textureSize, weightsTexture, maxWeightsTexture} = this.state;
 
-    const weightsTransformShaders = this.getShaders('weights-transform');
+    const weightsTransformShaders = this.getShaders({vs: weightsVs, fs: weightsFs});
     this._createWeightsTransform(weightsTransformShaders);
 
-    const maxWeightsTransformShaders = this.getShaders('max-weights-transform');
-    const maxWeightTransform = new TextureTransform(this.context.device, {
+    const maxWeightsTransformShaders = this.getShaders({vs: maxVs, fs: maxFs});
+    const maxWeightTransform = new TextureTransform(device, {
       id: `${this.id}-max-weights-transform`,
-      _sourceTextures: {
+      // TODO(donmccurdy): `inTexture` is a vertex attribute in the `max` shader, not a texture.
+      // That hasn't changed, and I'm not currently able to follow how this worked before...
+      // Luma is understandably warning about 'Unknown binding "inTexture"'.
+      bindings: {
```

Comment:
- Discussion notes: Possibly previous Transform was re-interpreting this input as a vertex attribute.